tests/vmi_multus_test.go

```diff
@@ -956,6 +976,20 @@ var _ = Describe("[Serial]Macvtap", func() {
 		return vmi
 	}
 
+	createCirrosVMIStaticIPRandomNode := func(virtClient kubecli.KubevirtClient, networkName string, mac string) *v1.VirtualMachineInstance {
+		return createRunningVMIFromSpec(
```

Comment:
The solution requires me to have a running VMI with a console so that I can ping / migrate / ping again.

The readability of each individual block is required, but - as you constantly say - providing the correct level of abstraction also is - e.g. if I need a "running cirros VMI with static IP with a console, not caring about which node it runs on", I should use a tiny wrapper that gives me that, rather than go and learn how to build one.

Reply:
Your claim here is that: `createCirrosVMIRandomNode("mynet", "ff:ff...")`
abstracts:
```
createRunningVMIFromSpec(newCirrosVMIWithExplicitMac("mynet, "ff:ff..."), tests.LoginToCirros)
```

Sorry, I do not see the abstraction here. Both functions you created in this scope and to me it does not read better.
If I look back on the work triggering the `libvmi`, it has exactly tried to avoid this avalanche by building the config in a flat manner and not in a stack of helpers that wrapped options and added them to the name.
The only difference here is that it is done in a smaller scope which is easier to handle.

Anyway, I won't block on this one, it may be a matter of taste.